config.py

```diff
@@ -432,42 +432,59 @@ def ignore(self, addr):
         '''Read self.IGNORE_FILE to check if addr (an IP) is in it.
 
         The file should consist of blank-separated addresses or address ranges
-        in CIDR format to ignore.'''
-        bits = inet_pton(valid_addr(addr), addr)
+        in CIDR format to ignore, with optional port or port-range.
+        '''
+        bits = inet_pton(valid_addr(addr.host), addr.host)
         try:
             with open(self.IGNORE_FILE) as ignore:
                 for line in ignore:
                     # comments
                     if line.lstrip().startswith('#'):
                         continue
                     for word in line.split():
+                        try:
+                            # is there a port or port range?
+                            host, port = word.split(':', 1)
+                            try:
```

Comment:
instead of try/except, consider searching for a "-" and a ":" before trying to split. I think it will lead to less nested try/excepts making flow much easier to follow...

something like:
```
if ':' in word:
  host, port = word.split(':', 1)
  if '-' in port:
    high, low = port.split('-', 1)
    try:
      high, low = int(high), int(low)
    except ValueError:
       print("error parsing whatever")
       continue
```

Reply:
Done.